shared/schema-mappers.ts

```
@@ -0,0 +1,366 @@
+    enableTransitionalSupport?: boolean;
+    deprecationWarnings?: boolean;
+    exclude?: string[];
+    omit?: (keyof TTable['_']['columns'])[];
+  } = {}
+) {
+  const { enableTransitionalSupport = true, deprecationWarnings = true, exclude = [], omit = [] } = options;
+  
+  // Get the base Drizzle schema with omissions
+  let baseSchema = createInsertSchema(table);
+  if (omit.length > 0) {
+    const omitObj = omit.reduce((acc, key) => ({ ...acc, [key]: true }), {});
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)